src/kagglehub/clients.py

```diff
@@ -54,27 +55,31 @@
   {}
 """
 
-_cached_user_agent = None
-
 
 def get_user_agent() -> str:
-    global _cached_user_agent  # noqa: PLW0603
```

Comment:
I would keep caching at least the ouput of `read_kaggle_build_date`. This requires reading a file and I would like to avoid having to read it for each network call.